superset/views/geocoding.py

```diff
@@ -230,200 +223,174 @@ def geocode(self) -> Response:
+        :return: The from clause as a string
+        """
         quote = table.database.get_sqla_engine().dialect.identifier_preparer.quote
         if table.schema:
             full_table_name = quote(table.schema) + "." + quote(table.table_name)
         else:
             full_table_name = quote(table.table_name)
         return text(full_table_name)
 
-    def _check_and_create_columns(self, request_data):
-        lat_column = request_data.get("latitudeColumnName", "lat")
-        lon_column = request_data.get("longitudeColumnName", "lon")
-        if_exists = request.json.get("overwriteIfExists", "fail")
-        if "fail" in if_exists:
-            override_if_exist = False
-        else:
-            override_if_exist = True
-        table_dto = request_data.get("datasource", models.TableDto())
-        table_id = table_dto.get("id", "")
-        table = self._get_table_by_id(table_id)
-        lat_exists = self._does_column_name_exist(table, lat_column)
-        lon_exists = self._does_column_name_exist(table, lon_column)
-        column_message_suffix = ""
-        if override_if_exist:
-            if lat_exists:
-                if lon_exists:
-                    pass
-                else:
-                    self._add_lat_lon_columns(table, lon_column=lon_column)
-                    column_message_suffix = f"successfully created column {lon_column}"
-            else:
-                if lon_exists:
-                    self._add_lat_lon_columns(table, lat_column=lat_column)
-                    column_message_suffix = f"successfully created column {lat_column}"
-                else:
-                    self._add_lat_lon_columns(table, lat_column, lon_column)
-                    column_message_suffix = (
-                        f"successfully created columns: {lat_column} and {lon_column}"
-                    )
-        else:
-            if lat_exists:
-                raise ValueError(
-                    "Column name {0} for latitude is already in use".format(lat_column)
-                )
-            if lon_exists:
-                raise ValueError(
-                    "Column name {0} for longitude is already in use".format(lon_column)
-                )
-            self._add_lat_lon_columns(table, lat_column, lon_column)
-            column_message_suffix = (
-                f"successfully created columns: {lat_column} and {lon_column}"
+    def _create_columns(
+        self, lat_column: str, lat_exists: bool, lon_column: str, lon_exists: bool, table: SqlaTable
+    ) -> None:
+        """
+        Check and create the lat / lon columns if needed
+        :param if_exists: What should happen if the columns already exist
+        :param lat_column: The name of the latitude column
+        :param lon_column: The name of the longitude column
+        :param table_id: The id of the table
+        :return: A message describing what columns have been created or an empty message if no columns have been created
```

Comment:
there is no param if_exists (not needed, remove from docstring), params lat_exists and lon_exists are not mentioned in docstring
docstring mentions table_id but signature contains table